build-tools/xa-prep-tasks/Xamarin.Android.BuildTools.PrepTasks/Which.cs

```diff
@@ -41,25 +41,26 @@ public static string GetProgramLocation (string programBasename, out string file
 		{
 			directories = directories ?? GetPathDirectories ();
 			foreach (var d in directories) {
-				var p   = GetProgramLocation (programBasename, d, out filename);
-				if (p != null)
-					return p;
+				foreach (var ext in FileExtensions) {
+					// Try lower case extension, %PATHEXT% is all caps on Windows
+					var p = GetProgramLocation (programBasename, d, ext?.ToLowerInvariant (), out filename);
```

Comment:
A "problem" here is that it means `.ToLowerInvariant()` is always called at least once per program lookup, which might not matter *that* much (`Path.ChangeExtension()` also creates strings), but it seems...less than ideal, particularly since this value won't actually be changing.

I think it would be better to instead "insert" the values into `FileExtensions` within the static constructor:

```csharp
FileExtensions  = new string [((pathExts?.Length ?? 0)*2) + 1];
for (int i = 0, j = 0; i < pathExts?.Length ?? 0 ; i++, j += 2) {
	FileExtensions [j] = pathExts [i].ToLowerInvariant ();
	FileExtensions [j+1] = pathExts [i];
}
FileExtensions [FileExtensions.Length - 1] = null;
```

This allows us to call `.ToLowerInvariant()` *once* per file extension, not multiple times per filename probe.